### PR TITLE
TFAC-588 Generalizing Onboarding Flow

### PR DESCRIPTION
**Pull request by jedtan**

### CODE REVIEW

File: src/components/__tests__/OnboardingFlow.test.js [hunk 1]
```diff
@@ -2,46 +2,71 @@ import React from 'react'
...
+import adCat from 'src/assets/onboarding/adcat.svg'
+
+const getMockProps = () => ({
+  showMissionSlide: false,
+  onboarding: {
+    steps: [
+      {
+        imgName: 'cattabs',
+        title: 'Your tabs are doing great things',
+        subtitle: 'Now, every tab you open supports cats in need.',
```
Comment (kmjennison) [inline src/components/__tests__/OnboardingFlow.test.js:16]: Use markdown in tests and assert on Typography variants.

File: src/pages/index.js [hunk 2]
```diff
@@ -270,6 +270,15 @@ const getRelayQuery = async ({ AuthUser }) => {
             tabCount
             missionId
           }
+          cause {
```
[the remark this answers is not in the document]
Comment (jedtan) [reply src/pages/index.js:273]: Do you mean a fragment?

File: src/components/OnboardingFlow.js [hunk 3]
```diff
@@ -50,85 +50,37 @@ export const useStyles = makeStyles((theme) => ({
...
-          redistribute that money to charity. No fees or hidden costs!
-        </Typography>
-      ),
-    },
-  ]
+  const onboardingStepContents = steps.map((step) => ({
+    imageSrc: getImageAssetFromName(step.imgName),
+    title: step.title,
+    children: (
+      <Typography align="center">
```
[the remark this answers is not in the document]
Comment (jedtan) [reply src/components/OnboardingFlow.js:75]: How would you recommend doing the centering?

File: src/components/__tests__/OnboardingFlow.test.js (hunk 1 above)
Comment (kmjennison) [reply src/components/__tests__/OnboardingFlow.test.js:16]: That works. In that case, maybe mock Markdown and assert it's receiving the prop value as children? Basically, looking to have a test of "this component knows how to use markdown".

File: src/components/OnboardingFlow.js [hunk 4]
```diff
@@ -138,18 +95,21 @@ const OnboardingFlow = ({ onComplete, showMissionSlide }) => {
...
     <Card className={classes.card}>
       <CardContent className={classes.cardContent}>
         <img
-          alt="cute cat"
+          alt="onboarding"
           className={classes.onboardingImage}
           src={onboardingStepInfo.imageSrc}
         />
-        <Typography variant="h5" className={classes.cardTitle}>
+        <Markdown className={classes.cardTitle}>
```
Comment (kmjennison) [inline src/components/OnboardingFlow.js:107]: `Markdown` doesn't actually use any other props, so I think this class is doing nothing. Just use a div?

File: src/components/OnboardingFlow.js [hunk 5]
```diff
@@ -138,18 +95,21 @@ const OnboardingFlow = ({ onComplete, showMissionSlide }) => {
...
           {onboardingStepInfo.title}
+        </Markdown>
+        <Typography align="center">
```
Comment (kmjennison) [inline src/components/OnboardingFlow.js:110]: Does using a div with `text-align: center` styling work? Nested `<p>` are technically against HTML spec.

**Review by kmjennison — APPROVED** [review]
LGTM! 🦊

**Pull request merged**